storage_v2/v1adapter/ptrace2model.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1adapter
+
+import (
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/iter"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	otel2model "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+)
+
+// PTracesSeq2ToModel consumes an iterator seqTrace. When necessary,
+// it groups spans from *consecutive* chunks of ptrace.Traces into a single model.Trace
+//
+// Returns nil, and spanstore.ErrTraceNotFound for empty iterators
+func PTracesSeq2ToModel(seqTrace iter.Seq2[[]ptrace.Traces, error]) ([]*model.Trace, error) {
+	var (
+		err         error
+		lastTraceID *model.TraceID
+		lastTrace   *model.Trace
+	)
+	jaegerTraces := []*model.Trace{}
+
+	seqTrace(func(otelTraces []ptrace.Traces, e error) bool {
```

Comment:
this implementation is correct, but you could simplify it by using `jptrace.AggregateTraces()` that already does the grouping, such that you only need to perform a conversion of single `ptrace.Traces` to a single `model.Trace`

Reply:
Alright